analysis/quantifier-comparisons/14-cellhash-exploration.Rmd

```diff
@@ -0,0 +1,371 @@
+
+Overall, I am not confident that the called cells are very reliable based on these data. 
+
+
+Let's pull out the identities for comparison later.
+```{r}
+seurat_classification <- data.frame(
+  cell_barcode = colnames(seurat_obj),
+  seurat_id = as.character(seurat_obj$hash.ID),
+  seurat_maxid = as.character(seurat_obj$HTO_maxID),
+  seurat_margin = seurat_obj$HTO_margin
+)
+```
+```{r}
+dplyr::count(seurat_classification, seurat_id)
+```
+
+```{r}
+ggplot(seurat_classification, aes(x = seurat_margin)) +
+  geom_histogram(bins = 50)+ 
+  labs(x = "Seurat Margin")
+```
+
+# Expression data
+
+Let's just look quickly at the expression data to see if that looks as we would expect.
+We will plot some basic cell metrics, including mitochondrial percent.
+
+```{r}
+cell_metrics <- colData(sce_filtered)|>
+  as.data.frame()|>
+  tibble::rownames_to_column("barcode")
+ggplot(cell_metrics, aes(x = sum, y = detected, color = subsets_mito_percent)) +
+  geom_point(size = 0.2) +
+  theme_bw() +
+  scale_x_log10() + 
+  scale_y_log10() +
+  scale_color_continuous(type = "viridis", limit = c(0,100)) + 
+  labs(x = "Total UMI count",
+       y = "Genes detected",
+       color = "Mito %") 
+```
+
+Aside from having lots of cells, that looks pretty normal.
+Let's add some more aggressive filtering for now.
+
+```{r}
+pass_barcodes <- cell_metrics |>
+  dplyr::filter(detected > 500,
+                subsets_mito_percent < 10) |>
+  dplyr::pull(barcode)
+sce_filtered <- sce_filtered[, pass_barcodes]
+```
+
+
+Now we will perform normalization:
+
+```{r}
+qclust <- scran::quickCluster(sce_filtered)
+sce_filtered <- sce_filtered |> 
+  scran::computeSumFactors(clusters = qclust) |>
+  scater::logNormCounts()
+```
+
+And some dimension reduction with UMAP:
+```{r}
+sce_filtered <- sce_filtered |>
+  scater::runPCA() |>
+  scater::runUMAP()
+```
+
+Plot the UMAP results:
+```{r}
+scater::plotUMAP(sce_filtered, point_size = 0.1)
+```
+There does seem to be some normal-ish looking structure there, but of course without any context, there is little to interpret!
+Are the clusters different cell types or different samples?
+
+### Add in the classification info
+
+Let's merge in all of the classification info we have from earlier:
+
+```{r}
+# make sure order is correct in classification table
+hash_class <- data.frame(cell_barcode = colnames(sce_filtered)) |>
+  dplyr::left_join(hashdrops_classification, by = "cell_barcode")|>
+  dplyr::left_join(seurat_classification, by = "cell_barcode") |>
+  dplyr::mutate(seurat_id = ifelse(seurat_id %in% c("Doublet","Negative"), NA_character_, seurat_id))
+
+# add columns to sce
+sce_filtered$seurat_id = hash_class$seurat_id
+sce_filtered$hashdrops_id = hash_class$hashdrops_id
+```
+
+Plot the UMAP results with classifications:
+```{r}
+### seurat 
+scater::plotUMAP(sce_filtered, point_size = 0.1, colour_by = "seurat_id")
+```
```

Comment:
Can you make this same plot for the `emptyDrops` version?

Reply:
I am assuming you mean Hashdrops here, and yes I can, but as you will see, it is not very informative with so few called cells!